frontend/src/app/components/crm/leads/leads.component.spec.ts

```diff
@@ -33,4 +33,22 @@ describe('LeadsComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should listen to selectors and be disabled', () => {
+    const rootElement = fixture.nativeElement;
+    const submitButton = rootElement.querySelector('button#emailThem') as HTMLButtonElement;
+    const checkboxes = Array.from<HTMLInputElement>(
+      rootElement.querySelectorAll('input[type=checkbox]'));
+   
+      
+      expect(submitButton.disabled).toBe(true); // disabled
+      fixture.detectChanges(); // angular change detection, sometimes you need more than this for observables
```

Comment:
You should check out Angular Material [harnesses](https://material.angular.io/cdk/test-harnesses/overview). They make testing easier and cleaner.

Reply:
will check it out